src/openfermion/hamiltonians/_hubbard.py

```diff
@@ -218,21 +225,20 @@ def fermi_hubbard(x_dimension, y_dimension, tunneling, coulomb,
 
                 # Add hopping term.
                 operators = ((site, 1), (bottom_neighbor, 0))
-                hopping_term = FermionOperator(operators, -tunneling)
-                hubbard_model += hopping_term
-                hubbard_model += hermitian_conjugated(hopping_term)
+
             else:
                 # Add hopping term.
                 operators = ((up_index(site), 1),
                              (up_index(bottom_neighbor), 0))
                 hopping_term = FermionOperator(operators, -tunneling)
                 hubbard_model += hopping_term
                 hubbard_model += hermitian_conjugated(hopping_term)
+
                 operators = ((down_index(site), 1),
                              (down_index(bottom_neighbor), 0))
-                hopping_term = FermionOperator(operators, -tunneling)
-                hubbard_model += hopping_term
-                hubbard_model += hermitian_conjugated(hopping_term)
 
-    # Return.
-    return hubbard_model
+            hopping_term = FermionOperator(operators, -tunneling)
+            hubbard_model += hopping_term
+            hubbard_model += hermitian_conjugated(hopping_term)
+
+    return normal_ordered(hubbard_model)
```

Comment:
Is there a particular reason you want to return a normal-ordered operator? I think normal-ordering should generally be left up to the user to perform. In this case all it does is change the ordering within the coulomb terms, and if someone is examining the returned operator, it may be helpful for it to match the ordering used in the definition of the model.

Reply:
This is something for discussion - my belief is that it's OK to normal order when there's very little overhead in doing so. The thing I don't like is having to wrap `normal_ordered()` around every call to a Hamiltonian, which feels like a lot of unnecessary duplication.

My feeling is that the tradeoff is between the user's ability to have control and not be forced to do expensive operations unnecessarily, while also keeping the code simple - instead of having to call the same two functions together every time, simplifying it into a single function. Thoughts? @babbush ?

Reply:
I'd rather you didn't put the normal ordered call here. The user should ask for that explicitly if it's something they want.